.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,26 @@
+version: '3.8'
+
+services:
+  app:
+    build: 
+      context: .
+      dockerfile: Dockerfile
+      args:
+        VARIANT: 14-buster
+    volumes:
+      - ..:/workspace:cached
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:db
+  db:
+    image: mongo:latest
+    restart: unless-stopped
+    volumes:
+      - mongodb-data:/data/db
+    environment:
+     MONGO_INITDB_ROOT_USERNAME: root
+     MONGO_INITDB_ROOT_PASSWORD: password
+     MONGODB_URI: mongodb://root:password@mongo:27017/
```

Comment:
These are in the correct spot

Reply:
I'm talking specifically about the MONGODB_URI env variable. That needs to be with the app container so it knows how to connect to Mongo. Also, I think the URI there is wrong since you named the service "db" and not "mongo".